### ISSUE TITLE
Station with multiple stops

### ISSUE THREAD

**Issue by quenthal** [Mallonbacka/custom-component-digitransit#25]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/ludeeus/integration_blueprint/issues?q=is%3Aissue+label%3A%22Feature+Request%22+).

### Is your feature request related to a problem? Please describe.

As stated, current limitation is that stations are not currently supported, only stops. This means you can't monitor all departures from a station with multiple stops or platforms.

However, is there way to monitor single track from train station? For example many small train stations have only two platforms, A and B, track 1 and 2, or track 3 and 4.  Can these be somehow appended to stop code?

### Describe the solution you'd like

While waiting for release, possibility to monitor single stops/platforms from train stations.

### Describe alternatives you've considered

-

### Additional context

-

**Comment by Mallonbacka** [comment]
It's possible to monitor a single track - they have stop codes, just the same as bus and tram stops do.

For example, if I search for 'Pasila' on the HSL homepage, I see a list which includes all of these individual stops. Here, the stop codes are visible, for example, track 2 has the stop code 'H0086'. It can be added using this integration, just like any other stop. 

![Screenshot of a list of train tracks, each with ](https://github.com/Mallonbacka/custom-component-digitransit/assets/79469789/0d2b2f77-bda9-4169-870f-5a72f44566dd)

You can also find them by zooming right in on the map and clicking a specific track. 

I'll close this issue now, but feel free to comment if something is still uncler.

**Comment by Mallonbacka** [comment]
Aha, I wasn't aware that any stations contain stops with the same stop code, that's annoying. I'll dig into this a little more and see if there is a quick workaround, if not, I'll try and make some progress on the station support soon.

**Comment by Mallonbacka** [comment]
Hello @quenthal ,
I've released a new version (v0.2.0) which allows you to add a stop either by stop code (the current way) or with the GTFS ID. Finding the GTFS ID is a bit more involved, but as far as I can tell, these should always be unique. 

For Tapanila station, track 3 has the GTFS ID `HSL:1392501` and track 4 is `HSL:1392551`. I hope this helps!

**Comment by quenthal** [comment]
Thanks! Using GTFS ID allows these to be added, however sensor stays without value, and I notice also this in the log:
Traceback (most recent call last):

`  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 258, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 414, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 168, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 492, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1009, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1132, in _async_write_ha_state
    state, attr, capabilities, shadowed_attr = self.__async_calculate_state()
                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1067, in __async_calculate_state
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1015, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 543, in state
    value = self.native_value
            ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/digitransit/sensor.py", line 46, in native_value
    nextDeparture = self.coordinator.data.get("data").get("stop").get("stoptimesWithoutPatterns")[0]`
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range

There might be some other issues in my installation, need to fiddle with it..

**Comment by Mallonbacka** [comment]
Thanks for the information. I'm struggling to get any information from the API about train departures, so this looks a bit more complex than I was imagining. It returns a valid Stop, but an empty list for `stoptimesWithoutPatterns`.

While I hadn't prepared for it yet, an empty `stoptimesWithoutPatterns` should set the sensor to Unknown, instead of causing a fatal error in the integration, so I've fixed that with a new version (v0.2.1). There are no new features, only error handling, but it should avoid logging these errors if the response is valid, but there are no departures. 

I'll make notes about how to get the upcoming train departures in issue #33, but I'm currently not able to estimate how much work this will require, as I didn't find the root cause yet.
